qiime/sdk/tests/test_method.py

```diff
@@ -374,6 +376,93 @@ def test_async(self):
             self.assertEqual(result.view(list),
                              [10, 20, 0, 42, 43, 99, -22, 55, 1])
 
+    def test_markdown_input_section_validation(self):
+        temp_dir = tempfile.TemporaryDirectory(prefix="qiime2-test-temp-")
+        root_dir = temp_dir.name
+        fp = os.path.join(root_dir, "bad_markdown.md")
+
+        with open(fp, mode="w") as fh:
+            fh.write("""---
+name: Bad markdown
```

Comment:
Can you put each of the invalid markdown strings into files under `qiime/sdk/tests/data/`? You can then access those files in your tests using:

```python
fp = pkg_resources.resource_filename('qiime.sdk.tests', 'data/my-markdown-file.md')
```

See [this test case](https://github.com/qiime2/qiime2/blob/master/qiime/sdk/tests/test_artifact.py#L256-L257) for an example.